### PR TITLE
refresh the file plural.pyx

### PR DESCRIPTION
**Pull request by fchapoton**
The main goal is to get rid of `ParentWithGens` there, replaced by `Ring`.

I also took the opportunity to clean cython-lint warnings and other code and doc details.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.

### CODE REVIEW

File: src/sage/rings/polynomial/plural.pyx [hunk 1]
```diff
@@ -339,17 +334,15 @@ cdef class NCPolynomialRing_plural(Ring):
         self._ngens = n
         self._term_order = order
 
-        Ring.__init__(self, base_ring, names, category=category)
+        Ring.__init__(self, base_ring, names, category=category,
+                      normalize=False)
```
Comment (mkoeppe) [inline src/sage/rings/polynomial/plural.pyx]: why this change? (perhaps add a comment)

File: src/sage/rings/polynomial/plural.pyx [hunk 2]
```diff
@@ -2721,7 +2716,7 @@ cdef class NCPolynomial_plural(RingElement):
             return (<NCPolynomialRing_plural>self._parent)._zero_element
 
         return new_NCP((<NCPolynomialRing_plural>self._parent),
-                                           p_Head(self._poly,(<NCPolynomialRing_plural>self._parent)._ring))
+                       p_Head(self._poly,(<NCPolynomialRing_plural>self._parent)._ring))
```
Comment (mkoeppe) [inline src/sage/rings/polynomial/plural.pyx]:
```suggestion
                       p_Head(self._poly, (<NCPolynomialRing_plural>self._parent)._ring))
```

**Review by mkoeppe — COMMENTED** [review]
Otherwise LGTM

**Comment by fchapoton** [comment]
voilà (not tested)

**Comment by mkoeppe** [comment]
LGTM. Doctest failures are unrelated

**Review by mkoeppe — APPROVED** [review]
